token/program-2022/src/processor.rs

```diff
@@ -757,6 +792,19 @@ impl Processor {
                     )?;
                     extension.authority = new_authority.try_into()?;
                 }
+                AuthorityType::TransferHook => {
```

Comment:
`TransferAuthority` is too general, and could pertain to an account delegate. And it's evolved past just doing permissioning, since you can really do anything in that hook.

How about `TransferHookProgramId`, since that's exactly what this authority controls?

Reply:
cool, I think that sounds great! Lets people know it should be a program.